### ISSUE TITLE
Formatter incompatibility: breaking multiple context managers

### ISSUE THREAD

**Issue by charliermarsh** [astral-sh/ruff#7441]
Given:

```python
if True:
    with tempfile.TemporaryDirectory() as d1:
        symlink_path = Path(d1).joinpath("testsymlink")
        with tempfile.TemporaryDirectory(dir=d1) as d2, tempfile.TemporaryDirectory(
            dir=d1
        ) as d4, tempfile.TemporaryDirectory(dir=d2) as d3, tempfile.NamedTemporaryFile(
            dir=d4
        ) as source_file, tempfile.NamedTemporaryFile(
            dir=d3
        ) as lock_file:
            pass
```

Black formats as:

```python
if True:
    with tempfile.TemporaryDirectory() as d1:
        symlink_path = Path(d1).joinpath("testsymlink")
        with tempfile.TemporaryDirectory(dir=d1) as d2, tempfile.TemporaryDirectory(
            dir=d1
        ) as d4, tempfile.TemporaryDirectory(dir=d2) as d3, tempfile.NamedTemporaryFile(
            dir=d4
        ) as source_file, tempfile.NamedTemporaryFile(
            dir=d3
        ) as lock_file:
            pass
```

Ruff formats as:

```python
if True:
    with tempfile.TemporaryDirectory() as d1:
        symlink_path = Path(d1).joinpath("testsymlink")
        with tempfile.TemporaryDirectory(dir=d1) as d2, tempfile.TemporaryDirectory(
            dir=d1
        ) as d4, tempfile.TemporaryDirectory(dir=d2) as d3, tempfile.NamedTemporaryFile(
            dir=d4
        ) as source_file, tempfile.NamedTemporaryFile(dir=d3) as lock_file:
            pass
```

**Comment by zanieb** [comment]
Hm it'd be great if this formatted as
```python
if True:
    with tempfile.TemporaryDirectory() as d1:
        symlink_path = Path(d1).joinpath("testsymlink")
        with (
            tempfile.TemporaryDirectory(dir=d1) as d2,
            tempfile.TemporaryDirectory(dir=d1) as d4,
            tempfile.TemporaryDirectory(dir=d2) as d3,
            tempfile.NamedTemporaryFile(dir=d4) as source_file,
            tempfile.NamedTemporaryFile(dir=d3) as lock_file,
        ):
            pass
```

which Ruff did for me once but I'm not sure how I triggered it, perhaps with an old version.

**Comment by MichaReiser** [comment]
> which Ruff did for me once but I'm not sure how I triggered it, perhaps with an old version.

I think this only happens if you parenthesize the with item**s** and not each individual with item.

**Comment by MichaReiser** [comment]
IMO: Splitting all calls just because a context manager splits seems unnecessary.

**Comment by charliermarsh** [comment]
We're gonna call this a known deviation that we don't plan to fix. When we implement preview style, we'll support adding parentheses here for supported versions (see: https://github.com/astral-sh/ruff/issues/7234#issuecomment-1720273440), which will achieve the formatting that @zanieb suggested.

**Comment by charliermarsh** [comment]
Can be closed once we add documentation.

**Comment by charliermarsh** [comment]
(I don't think was closed by #7694; I _think_ that PR closed a different issue.)
